### PR TITLE
Issue#17 - Shopping Cart

### LINKED ISSUE #17: [Order Process] Cart Panel UI
## User Story
As a **User**, I would like **to be able to view all the items currently in my cart and a preview of their picture, name, price, quantity, and special instructions** so that **I can keep track of the Dishes I am ordering**.

## Detailed Description
Be able to access my shopping cart from the major navigation screens of the Potluck app, and see what items I have added to that cart. Also, I should be able to checkout from viewing my cart so that I can order the dishes that I want to. This user experience should be streamlined and easy to navigate. In addition, I should be able to edit my shopping cart to increase/decrease the quantity of items that I want to order.

## Acceptance Criteria
- [x] Given **an empty cart**, then **there should be an indication that nothing is in the cart yet.**
- [x] Given **items in the cart**, then **their quantities can be dynamically updated, and I can see the new total prices.**
- [x] Given **I can navigate to the dish pages**, then **I can add these dishes to my cart.**
- [x] Given **I add the same dish multiple times to my cart**, then **there should not be any duplicate entries, only the quantities should be updated.**
- [x] Given **there are items in my cart**, then **I should see an indication of that from the featured menu.**
- [x] Given **I click the "Add to Cart" button from the dish page**, then **I should know that the item was added successfully.**

## Tasks
- [x] Create Cart screen/page
- [x] Connect this to the "Add to Cart" button on each dish page
- [x] Link each item in the cart back to their respective dish pages
- [x] Increase/decrease quantity from shopping cart page
- [x] Create shopping cart cards for clearer communication
- [x] Add compatibility with database updates/information

### LINKED ISSUE #18: [Order Process] Add Dish to Cart Interface
## User Story
As a **User**, I would like **to be able to add a Dish to my cart from the Dish Page** so that **I can move forward with my order when I choose a Dish out**.

## Detailed Description
Add dish to cart directly from the dish page. Grouped with Issue #17 , closed with #63. 

## Acceptance Criteria
- [X] All acceptance criteria for #17

### PR DESCRIPTION
**Pull request by shikham-8**
Closes #17 and closes #18

Adds Shopping Cart screen, navigation, and related components (i.e. card). 

General structure:
- ShoppingCart component - renders cart component with cards based on what items have been added to a single shopping cart
- Shopping Cart updates are propagated throughout all other calculations that use them and displayed on the cart cards
- Badge on cart icon indicates items in the cart
- Shopping Cart saves contents and quantities
- Also added alert from dish page to indicate item was added to cart successfully

Screenshots:
image
image
image
image

### CODE REVIEW

File: app/screens/DishPage.js [hunk 1]
```diff
@@ -1,342 +1,435 @@
...
-import { color } from 'react-native-reanimated';
-import { SafeAreaView } from 'react-native-safe-area-context';
-import Carousel, {Pagination} from 'react-native-snap-carousel';
-import Reviews from '../components/Reviews';
-import colors from '../config/colors';
-import { getNDishReviews } from '../util/Queries';
-
-const SLIDER_WIDTH = Dimensions.get('window').width
-const ITEM_WIDTH = Math.round(SLIDER_WIDTH)
+import React, { useEffect, useState } from "react";
```
Comment (shikham-8) [inline app/screens/DishPage.js:1]: Changes on this page are mostly formatting (linter)

File: app/screens/DishPage.js [hunk 2]
```diff
@@ -1,342 +1,435 @@
...
+                if (found < 0) {
+                  const dish = props.Dish;
+                  addToCart({ dish, count });
+                } else {
+                  addQuantity(found, count);
+                }
+                Alert.alert(
+                  "Added to Cart",
+                  "Successfully added items to cart!"
+                );
```
Comment (shikham-8) [inline app/screens/DishPage.js:215]: Added the above onPress functionality to add a dish to the cart and indicate with an alert

**Comment by brendon-ng** [comment]
Looks good!
tiny change, could we make the badge next to the shopping cart the green in colors.secondary? if it looks good

**Review by jcgubbi — APPROVED** [review]

**Review by armada877 — APPROVED** [review]

**Review by brendon-ng — APPROVED** [review]
LGTM

**Pull request merged**